test/index.js

```diff
@@ -0,0 +1,18 @@
+var R = require('..');
+var fs = require('fs');
+var path = require('path');
+var eq = require('./shared/eq');
+
+
+describe('api surface', function() {
+  var srcFolder = path.join(__dirname, '../src');
+  var isJsFile = function(file) { return file.match(/\.js$/); }
+  var removeJsEnding = function(file) { return file.replace('.js', ''); }
+
+  var exportedApi = Object.keys(R);
+  var actualApi = fs.readdirSync(srcFolder).filter(isJsFile).map(removeJsEnding);
```

Comment:
> It forces us to make assumptions about the project layout. Were we to add src/somePrivateFunction.js our assertion would fail because R would not have a property named somePrivateFunction.

will tell more on that in half an hour